### DISCUSSION TITLE
Is this different Pickle from Python lib?

### DISCUSSION THREAD

**Discussion by taehalim** [apple/pkl#135] [Q&A] [answered]
This has the same file format "pkl" as the python module called pickle. 

I'm confused 😕 

Or has it been already integrated with Python years back already? 

Sending help.

**Comment by bioball** [comment, marked as answer]
Yes, this is not the same thing as Python's pickle. Python's pickle is about serializing/deserializing Python objects, and our is a configuration language.

Our file extension (`.pkl`) can conflict, although as far as I'm aware, that's due to convention. Python's pickle files don't have an extension, and I've seen `.p`, `.pkl`, and `.pickle` in the wild.

**Reply by taehalim** [reply]
Thanks. Everything is clear now. But why did you guys name it pickle? Just out of curiosity 🥒

**Reply by holzensp** [reply]
Mostly historical, referring to organisation names that no long exist and then culture that grew. It's certainly a wink to [Tcl](https://www.tcl.tk).
